src/couch/src/couch_doc.erl

```diff
@@ -126,8 +128,11 @@ doc_to_json_obj(#doc{id=Id,deleted=Del,body=Body,revs={Start, RevIds},
     }.
 
 from_json_obj_validate(EJson) ->
+    from_json_obj_validate(EJson, <<"">>).
```

Comment:
`undefined` atom is a safer default, I can see bugs if we allow a database name to be the empty string even in this case.